formats/json/jvmTest/src/kotlinx/serialization/features/JsonStreamFlowTest.kt

```diff
@@ -36,16 +46,116 @@ class JsonStreamFlowTest {
             f.writeToStream(os)
         }
 
-        assertEquals(inputString, os.toString(Charsets.UTF_8.name()))
+        assertEquals(inputStringWsSeparated, os.toString(Charsets.UTF_8.name()))
     }
 
     @Test
     fun testDecodeSeveralItems() {
-        val ins = ByteArrayInputStream(inputString.encodeToByteArray())
+        val ins = ByteArrayInputStream(inputStringWsSeparated.encodeToByteArray())
         assertFailsWithMessage<SerializationException>("EOF") {
             json.decodeFromStream<StringData>(ins)
         }
     }
 
+    private inline fun <reified T> Iterator<T>.assertNext(expected: T) {
```

Comment:
Seems like this functionality might have slightly more tests. WDYT about adding `JsonTestingMode` for streams?
AUTO and WS can be tested as is, and for `ARRAY` we can just wrap the strings into `[]`

Reply:
I don't think it will give benefits. Underlying parsing logic is the same as in `JsonTestingMode.JAVA_STREAMS`, the only difference we need to test is object separators. Since all of the tests with `JsonTestingMode` parse one object at a time, separators will likely be untested.